rasa/cli/arguments/train.py

```diff
@@ -57,7 +57,28 @@ def set_train_nlu_arguments(parser: argparse.ArgumentParser):
     add_persist_nlu_data_param(parser)
 
 
-def add_force_param(parser: Union[argparse.ArgumentParser, argparse._ActionsContainer]):
+def set_train_fingerprint_arguments(parser: argparse.ArgumentParser) -> None:
```

Comment:
Maybe @b-quachtran can shed some light on this:
* the current PR adds a command to create a fingerprint and share it. Users have to manage the persisted fingerprints manually and also compare them manually
* from what I understand, the underlying motivation for this feature is to determine whether a model should be re-trained. From that perspective I'd favor something like a `--dry-run` flag for `rasa train` because
  - less additional code
  - We can actually do the comparison for them as long as they have the last model in the `models` (`out` param) directory and e.g. give different exit codes + a description as output. E.g. exit code `0` means no training necessary, `1` means everything needs to be re-trained, `2` means only `core` needs training and `3` only NLU needs training (this case is gonna be removed with end-to-end)
   - There are some discussions between Ben and me in the [issue](https://github.com/RasaHQ/rasa/issues/6729#issuecomment-729168231) which also would make favor the `--dry-run` solution

what do you think?

Reply:
@wochinge  you mean something like `rasa train --only-if-needed` (the name is here only to illustrate the idea)? 

We might do that, either additionally to what's done here or instead of the fingerprint command. It's sort of confusing for me that the description of the issue itself is different but it's not a big deal — we can adapt the code here to the new conditions. However, I would like to also hear @b-quachtran's opinion on that.

Reply:
Nope, rather a `--dry-run` which never trains but only indicates if a training would be performed if the command would be run without `--dry-run

Reply:
`FingerprintComparisonResult.is_training_required` also returns true if the responses in the domain should be updated (`FingerprintComparisonResult.nlg` indicates that). Should we also return an error code in this case?
I drafted a PR for it (https://github.com/RasaHQ/rasa/pull/7397), just want to know your opinion before I can add tests and update the code style.

Reply:
> Should we also return an error code in this case?

Yes 👍

Reply:
@b-quachtran Could you please have a short look as well?

Reply:
@wochinge @joejuzl nothing is really happening with this PR but I already made a new one: https://github.com/RasaHQ/rasa/pull/7397 could you please check it as well? :)

Reply:
This would be for use within a CI pipeline so that a fingerprint can be outputted and diff-ed against previous fingerprints. This customer is running training outside of the runner and they want a way to validate if training needed ahead of that external call so `rasa train --only-if-needed` wouldn't work

Reply:
@wochinge so, do you think we should close this one in favour of https://github.com/RasaHQ/rasa/pull/7397?

Reply:
I'd say so. Or can you think of a good use case for it?